### PR TITLE
Course 6

### PR DESCRIPTION
**Pull request by AnnaMartis**

### CODE REVIEW

File: src/pages/MovieListPage/MoviesListPage.jsx [hunk 1]
```diff
@@ -1,41 +1,62 @@
...
   const [searchQuery, setSearchQuery] = useState("");
-  const [sortCriterion, setSortCriterion] = useState(moviesSortOptions[0]);
-  const [selectedGenre, setSelectedGenre] = useState("ALL");
+  const [sortCriterion, setSortCriterion] = useState(
+    moviesSortOptions.find(
+      (option) => option.value === searchParams.get("sort")
+    ) || moviesSortOptions[0]
+  );
+  const [selectedGenre, setSelectedGenre] = useState(
+    searchParams.get("genre") || "ALL"
```
Comment (dmitriisurdin) [inline src/pages/MovieListPage/MoviesListPage.jsx]:
```suggestion
    searchParams.get("genre") ?? "ALL"
```

File: src/pages/MovieListPage/MoviesListPage.jsx [hunk 2]
```diff
@@ -1,41 +1,62 @@
...
   const [movies, setMovies] = useState([]);
-  const [selectedMovie, setSelectedMovie] = useState(null);
 
   const [searchQuery, setSearchQuery] = useState("");
-  const [sortCriterion, setSortCriterion] = useState(moviesSortOptions[0]);
-  const [selectedGenre, setSelectedGenre] = useState("ALL");
+  const [sortCriterion, setSortCriterion] = useState(
+    moviesSortOptions.find(
+      (option) => option.value === searchParams.get("sort")
+    ) || moviesSortOptions[0]
```
Comment (dmitriisurdin) [inline src/pages/MovieListPage/MoviesListPage.jsx]:
```suggestion
      (option) => option.value === searchParams.get("sort")
    ) ?? moviesSortOptions[0]
```

File: src/pages/MovieListPage/MoviesListPage.jsx [hunk 3]
```diff
@@ -1,41 +1,62 @@
...
 
-  const handleMovieTitleClick = (movieProps) => {
-    setSelectedMovie(movieProps);
-  };
-
-  const handleMovieViewClosed = () => {
-    setSelectedMovie(null);
+  const handleMovieClick = (id) => {
+    const prevParams = Object.fromEntries(searchParams.entries());
+    navigate({
```
Comment (dmitriisurdin) [inline src/pages/MovieListPage/MoviesListPage.jsx:56]: why wouldn't we just use `<Link>` tag instead?

File: src/entities/MovieDetails/MovieDetails.jsx [hunk 4]
```diff
@@ -1,9 +1,50 @@
...
+    releaseYear: 2016 - 12 - 29,
+    imgUrl: "https://via.placeholder.com/300/92c952",
+    duration: "128",
+    rating: 7.9,
+    description:
+      "Mia, an aspiring actress, serves lattes to movie stars in between auditions and Sebastian, a jazz musician, scrapes by playing cocktail party gigs in dingy bars, but as success mounts they are faced with decisions that begin to fray the fragile fabric of their love affair, and the dreams they worked so hard to maintain in each other threaten to rip them apart.",
+  },
+}) => {
+  const [searchParams] = useSearchParams();
+  const navigate = useNavigate();
```
Comment (dmitriisurdin) [inline src/entities/MovieDetails/MovieDetails.jsx:22]: https://reactrouter.com/en/main/hooks/use-navigate

image

File: src/entities/MovieDetails/MovieDetails.jsx [hunk 5]
```diff
@@ -1,9 +1,50 @@
...
+
+  let imgUrl, movieName, releaseYear, rating, duration, description;
+
+  const data = useLoaderData();
+
+  if (data.message === "error") {
```
Comment (dmitriisurdin) [inline src/entities/MovieDetails/MovieDetails.jsx:28]: really strange approach to take data from the error  :)

File: src/entities/SearchForm/SearchForm.jsx [hunk 6]
```diff
@@ -20,15 +20,24 @@ export const SearchForm = ({ initialValue = "", onSearch, placeholder }) => {
...
+    >
       <input
         type="text"
         data-testid="search-input"
         className="search-input"
         value={searchValue}
         onChange={handleInputChange}
         onKeyDown={handleKeyPress}
-        onFocus={onSearch}
+        onFocus={(event) => {
```
Comment (dmitriisurdin) [inline src/entities/SearchForm/SearchForm.jsx]: I would move this logic to a separate handler

File: src/entities/SearchFormWrapper/SearchFormWrapper.jsx [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+  return (
+    <>
+      <div className="add-movie-part">
+        <CustomButton className="outlined-button" label="+ADD MOVIE" />
+      </div>
+      <div className="find-movie-part">
+        <p className="find-movie-title">FIND YOUR MOVIE</p>
+        <SearchForm
+          initialValue={searchParam || ""}
```
Comment (dmitriisurdin) [inline src/entities/SearchFormWrapper/SearchFormWrapper.jsx]:
```suggestion
          initialValue={searchParam ?? ""}
```

File: src/entities/SearchFormWrapper/SearchFormWrapper.jsx [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+          onSearch={handleSearch}
+          placeholder={"What do you want to search?"}
```
Comment (dmitriisurdin) [inline src/entities/SearchFormWrapper/SearchFormWrapper.jsx]:
```suggestion
          placeholder="What do you want to search?"
```

File: src/routes/movie.jsx [hunk 9]
```diff
@@ -0,0 +1,10 @@
+export const loader = async ({ params }) => {
+  try {
+    const response = await fetch(`http://localhost:4000/${params.movieId}`);
+    const json = await response.json();
```
Comment (dmitriisurdin) [inline src/routes/movie.jsx]:
```suggestion
    return await response.json();
```

File: src/pages/MovieListPage/MoviesListPage.jsx (hunk 1 above)
Comment (AnnaMartis) [reply src/pages/MovieListPage/MoviesListPage.jsx]: done

File: src/pages/MovieListPage/MoviesListPage.jsx (hunk 2 above)
Comment (AnnaMartis) [reply src/pages/MovieListPage/MoviesListPage.jsx]: done

File: src/pages/MovieListPage/MoviesListPage.jsx (hunk 3 above)
Comment (AnnaMartis) [reply src/pages/MovieListPage/MoviesListPage.jsx:56]: we have to write ` const prevParams = Object.fromEntries(searchParams.entries());`  this part manually, (we have to add a function which will evaluate those parts and put inside `<Link to="here"></Link>`). wouldn't it better to use "navigate"?

File: src/entities/MovieDetails/MovieDetails.jsx (hunk 5 above)
Comment (AnnaMartis) [reply src/entities/MovieDetails/MovieDetails.jsx:28]: in our case backend endpoint doesn't work, so I have tried to put a mock data inside error part

File: src/entities/MovieDetails/MovieDetails.jsx (hunk 4 above)
Comment (AnnaMartis) [reply src/entities/MovieDetails/MovieDetails.jsx:22]: don't they mean inside loader files ?

File: src/entities/SearchForm/SearchForm.jsx (hunk 6 above)
Comment (AnnaMartis) [reply src/entities/SearchForm/SearchForm.jsx]: done

File: src/entities/SearchFormWrapper/SearchFormWrapper.jsx (hunk 7 above)
Comment (AnnaMartis) [reply src/entities/SearchFormWrapper/SearchFormWrapper.jsx]: done

File: src/entities/SearchFormWrapper/SearchFormWrapper.jsx (hunk 8 above)
Comment (AnnaMartis) [reply src/entities/SearchFormWrapper/SearchFormWrapper.jsx]: done

File: src/routes/movie.jsx (hunk 9 above)
Comment (AnnaMartis) [reply src/routes/movie.jsx]: done

File: src/entities/MovieDetails/MovieDetails.jsx (hunk 5 above)
Comment (dmitriisurdin) [reply src/entities/MovieDetails/MovieDetails.jsx:28]: yeah, that's fine for our case

File: src/entities/MovieDetails/MovieDetails.jsx (hunk 4 above)
Comment (dmitriisurdin) [reply src/entities/MovieDetails/MovieDetails.jsx:22]: using `redirect` inside `loader` doesn't mean that, but you can keep it as is

File: src/entities/SearchForm/SearchForm.jsx [hunk 10]
```diff
@@ -19,16 +19,27 @@ export const SearchForm = ({ initialValue = "", onSearch, placeholder }) => {
...
+  const handleInputFocus = (event) => {
+    event.preventDefault();
+    onSearch(searchValue);
+  }
+
   return (
-    <form className="search" onSubmit={() => onSearch(searchValue)}>
+    <form
+      className="search"
+      onSubmit={(event) => {
```
Comment (dmitriisurdin) [inline src/entities/SearchForm/SearchForm.jsx]: same, let's move to a separate handler

**Review by dmitriisurdin — APPROVED** [review]

File: src/entities/SearchForm/SearchForm.jsx (hunk 10 above)
Comment (AnnaMartis) [reply src/entities/SearchForm/SearchForm.jsx]: done

Comment (AnnaMartis) [reply src/entities/SearchForm/SearchForm.jsx]: done

**Pull request merged**